price-aggregator/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+#![no_std]
+#![feature(destructuring_assignment)]
+
+elrond_wasm::imports!();
+pub mod median;
+mod price_aggregator_data;
+
+use elrond_wasm::require;
+use price_aggregator_data::{OracleStatus, PriceFeed, TokenPair};
+
+#[elrond_wasm_derive::contract]
+pub trait PriceAggregator {
+    #[init]
+    fn init(
+        &self,
+        payment_token: TokenIdentifier,
+        oracles: Vec<Address>,
+        submission_count: u32,
+        decimals: u8,
+        query_payment_amount: Self::BigUint,
+    ) -> SCResult<()> {
+        self.payment_token().set(&payment_token);
+        self.query_payment_amount().set(&query_payment_amount);
+        self.submission_count().set(&submission_count);
+        self.decimals().set(&decimals);
+        oracles.iter().for_each(|oracle| {
+            self.oracle_status().insert(
+                oracle.clone(),
+                OracleStatus {
+                    total_submissions: 0,
+                    accepted_submissions: 0,
+                },
+            );
+        });
+        Ok(())
+    }
+
+    #[endpoint]
+    #[payable("*")]
+    fn deposit(
+        &self,
+        #[payment] payment: Self::BigUint,
+        #[payment_token] token: TokenIdentifier,
+    ) -> SCResult<()> {
```

Comment:
I think it would be useful to have an optional argument: `on_behalf_of`, so payments can be delegated to the address provided, if any

Reply:
I added it.